### PR TITLE
feat(csrf): added functionality for cross site request forgery (csrf)

### PR DESCRIPTION
**Pull request by paddybasi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-1957

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/auth/models/strategy.class.ts [hunk 1]
```diff
@@ -182,6 +186,7 @@ export abstract class Strategy extends events.EventEmitter {
         this.initializePassport()
         this.initializeSession()
         this.initializeKeepAlive()
+        this.initialiseCSRF()
```
Comment (udaydenduluri33) [inline src/auth/models/strategy.class.ts:189]: Question - because the options are already can we add the check  if (this.options.useCSRF) here itself.

**Review by Daniel-Lam — APPROVED** [review]
Approved, @paddybasi

File: src/auth/models/strategy.class.ts (hunk 1 above)
Comment (udaydenduluri33) [inline src/auth/models/strategy.class.ts:189]:
```suggestion
 if (this.options.useCSRF) {
        this.initialiseCSRF()
        }
```

**Review by rayliangatos — APPROVED** [review]

File: src/auth/models/strategy.class.ts (hunk 1 above)
Comment (paddybasi) [reply src/auth/models/strategy.class.ts:189]: that is already part of the initialise function:
public initialiseCSRF = (): void => {
        if (this.options.useCSRF) {
            this.logger.log('initialising CSRF middleware')
            this.router.use(csrf())
        }
    }

Comment (udaydenduluri33) [reply src/auth/models/strategy.class.ts:189]: I know but inside the method initialiseCSRF. Inside vs outside was my question

**Review by udaydenduluri33 — APPROVED** [review]
LGTMp

**Pull request merged**